docs/language/informal/covariant-overrides.md

```diff
@@ -0,0 +1,742 @@
+supertype or a subtype of the type declared for the same parameter in each
+of the directly or indirectly overridden declarations.
+
+It is not enough to require a relationship with the directly overridden
+declarations: If we only required a subtype or supertype relation to each
+directly overridden declaration, we can easily create an example showing
+that there are no guarantees about the relationship between the statically
+known (original) parameter type and the actual (desired) parameter type
+at run time:
+
+```dart
+class A { void foo(int i) {...}}
+class B implements A { void foo(Object o) {...}}
+class C implements B { void foo(covariant String s) {...}}
+
+main() {
+  A a = new C();
+  a.foo(42); // Original type: int, desired type: String: Unrelated.
+}
+```
+
+Checking all overridden declarations may seem expensive, because all
+supertypes must be inspected in order to find those declarations. However, we
+expect that the cost can be kept at a reasonable level:
+
+First, covariant parameters are expected to be rare, and it is possible for
+a compiler to detect them at a reasonable cost: For each parameter, the status
+of being covariant or not covariant can be maintained by storing it after
+inspecting the stored value for that parameter in each directly overridden
+method. With that information available, it is cheap to detect whether a given
+method has any covariant parameters. If that is not the case then the override
+checks are unchanged, compared to the language without this feature.
+
+Otherwise the parameter is covariant, and in this case it will be necessary to
+find all overridden declarations (direct and indirect) for that method, and
+gather all types for that parameter. Given that overridden methods would
+themselves often have a corresponding covariant parameter, it may be worthwhile
+to cache the result. In that case there will be a set of types that occur as
+the declared type of the given parameter, and the check will then be to iterate
+over these overridden parameter types and verify that the overriding parameter
+type is a subtype or a supertype of each of them.
+
+#### Subtyping
+
+`covariant` modifiers are ignored when deciding subtype relationships among
+classes. So this:
+
+```dart
+var group = new RadioGroup();
+Widget widget = group;
+```
+
+... is perfectly fine.
+
+#### Method invocations
+
+Method invocations are checked according to the statically known receiver type,
+and all `covariant` modifiers are ignored. That is, we check against that which
+we called the original parameter types, and we ignore that the desired parameter
+type may be different. So this:
+
+```dart
+var group = new RadioGroup();
+group.addChild(new Widget()); // <--
+```
+
+... reports an error on the marked line. But this:
+
+```dart
+Widget widget = new RadioGroup();
+widget.addChild(new Widget());
+```
+
+... does not. Both will fail at run time, but the whole point of allowing
```

Comment:
The one that reports an error at compile time won't fail at runtime - since it can never run.